### ISSUE TITLE
Make `Procompose` polykinded

### ISSUE THREAD

**Issue by Icelandjack** [ekmett/profunctors#41]
``` haskell
Procompose :: (xxx -> k2 -> *) -> (k1 -> xxx -> *) -> k1 -> k2 -> *
```

instead of

``` haskell
Procompose :: (* -> * -> *) -> (* -> * -> *) -> * -> * -> *
```

There's probably other examples

**Comment by RyanGlScott** [comment]
Good idea. Pull requests welcome.

**Comment by RyanGlScott** [comment]
I believe this was implemented in #90. Please re-open if this is not the case.
